core/src/services/s3/core.rs

```diff
@@ -433,7 +433,7 @@ impl S3Core {
         }
 
         let mut req = req
-            .header(constants::X_AMZ_COPY_SOURCE, percent_encode_path(&source))
+            .header(constants::X_AMZ_COPY_SOURCE, &source)
```

Comment:
Thanks a lot! Would you like to add a behavior test case for us to avoid this breaking again?

Reply:
You can add a test case here: https://github.com/apache/incubator-opendal/blob/main/core/tests/behavior/write.rs

> NOTE: This PR is good enough for merging. It's okay if you don't want to do any additional work. Please let me know.

Reply:
OK, I would compliment it later this week, I want to add UT for it originally, but didn't know how at first, and thought it's mocked as to not coveraged.